source/common/network/listen_socket_impl.h

```diff
@@ -175,9 +175,14 @@ class AcceptedSocketImpl : public ConnectionSocketImpl {
 // ConnectionSocket used with client connections.
 class ClientSocketImpl : public ConnectionSocketImpl {
 public:
-  ClientSocketImpl(const Address::InstanceConstSharedPtr& remote_address)
+  ClientSocketImpl(const Address::InstanceConstSharedPtr& remote_address,
+                   const OptionsSharedPtr& options)
       : ConnectionSocketImpl(remote_address->socket(Address::SocketType::Stream), nullptr,
-                             remote_address) {}
+                             remote_address) {
+    if (options) {
```

Comment:
Huh, that's a strange lapse - nice catch.
Mind adding a regression test?

Reply:
Thanks for the quick feedback, will add a test ASAP :-)

Reply:
Regression test added.